presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -200,7 +201,16 @@ public static ReaderPageSourceWithProjections createPageSource(
             }
 
             Map<List<String>, RichColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, effectivePredicate);
+
+            Map<List<String>, RichColumnDescriptor> descriptorForTupleDomain;
+            if (useColumnNames) {
+                descriptorForTupleDomain = descriptorsByPath;
+            }
+            else {
+                descriptorForTupleDomain = getDescriptorsUsingIndex(columns, fileSchema);
```

Comment:
I think it would be more adequate to place the fix inside the `getParquetTupleDomain`. Did you try that?

Reply:
Moved 👍